visualization/app/codeCharta/ui/ribbonBar/linkHeightAndColorMetricButton/linkHeightAndColorMetricButton.component.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { render, screen } from "@testing-library/angular"
+import userEvent from "@testing-library/user-event"
+import { LinkHeightAndColorMetricButtonComponent } from "./linkHeightAndColorMetricButton.component"
+import { Store } from "../../../state/store/store"
+import { isHeightAndColorMetricLinkedSelector } from "../../../state/store/appSettings/isHeightAndColorMetricLinked/isHeightAndColorMetricLinked.selector"
+import { TestBed } from "@angular/core/testing"
+import { CommonModule } from "@angular/common"
+
+describe("linkHeightAndColorMetricComponent", () => {
+	beforeEach(() => {
+		Store["initialize"]()
+		TestBed.configureTestingModule({
+			imports: [CommonModule]
+		})
+	})
+
+	it("should toggle height and color metric link on click and show associated icon", async () => {
+		const { container } = await render(LinkHeightAndColorMetricButtonComponent)
+		const isHeightAndColorMetricLinked = isHeightAndColorMetricLinkedSelector(Store.store.getState())
+
+		expect(container.getElementsByClassName("fa fa-chain-broken")[0].className).toEqual("fa fa-chain-broken")
```

Comment:
I find `expect(container.querySelector(".fa.fa-chain-broken")).not.toBe(null)` more intuitive. In case the element is not there `container.getElementsByClassName("fa fa-chain-broken")` would be an empty array and the test would break with a  null pointer exception and not fail with an assertion.